packages/multi-test/src/app.rs

```diff
@@ -49,35 +58,115 @@ where
     }
 }
 
-impl<C> Executor<C> for App<C>
+impl<ExecC, QueryC> Executor<ExecC> for App<ExecC, QueryC>
 where
-    C: Clone + fmt::Debug + PartialEq + JsonSchema + 'static,
+    ExecC: Clone + fmt::Debug + PartialEq + JsonSchema + 'static,
+    QueryC: CustomQuery + DeserializeOwned,
 {
-    fn execute(&mut self, sender: Addr, msg: CosmosMsg<C>) -> AnyResult<AppResponse> {
+    fn execute(&mut self, sender: Addr, msg: CosmosMsg<ExecC>) -> AnyResult<AppResponse> {
         let mut all = self.execute_multi(sender, vec![msg])?;
         let res = all.pop().unwrap();
         Ok(res)
     }
 }
 
-impl<C> App<C>
+/// Utility to build App in stages. If particular items wont be set, defaults would be used
+#[derive(Derivative)]
+#[derivative(Default(bound = "", new = "true"))]
+pub struct AppBuilder<ExecC, QueryC> {
+    wasm: Option<Box<dyn Wasm<ExecC, QueryC>>>,
+    bank: Option<Box<dyn Bank>>,
+    api: Option<Box<dyn Api>>,
+    storage: Option<Box<dyn Storage>>,
+    custom: Option<Box<dyn CustomHandler<ExecC, QueryC>>>,
+    block: Option<BlockInfo>,
+}
+
+impl<ExecC, QueryC> AppBuilder<ExecC, QueryC>
 where
-    C: Clone + fmt::Debug + PartialEq + JsonSchema + 'static,
+    ExecC: Debug + PartialEq + Clone + JsonSchema + 'static,
+    QueryC: CustomQuery + DeserializeOwned + 'static,
 {
-    pub fn new(
-        api: impl Api + 'static,
-        block: BlockInfo,
-        bank: impl Bank + 'static,
-        storage: impl Storage + 'static,
-    ) -> Self {
+    /// Overwrites default wasm executor. Panic if already set.
+    #[track_caller]
+    pub fn with_wasm(mut self, wasm: impl Wasm<ExecC, QueryC> + 'static) -> Self {
+        assert!(self.wasm.is_none(), "Wasm executor already overwritten");
```

Comment:
We don't allow overriding it twice? I guess no need, but seems odd to enforce.

Reply:
If you overwrite something twice in your builder then you made some mistake. Better to catch early.